### ISSUE TITLE
macOS conda env: libiconv 1.15 can't be found in default conda repo

### ISSUE THREAD

**Issue by joefresna** [DiODeProject/MuMoT#195]
A user may get stuck when installation step 4 gives an error, e.g.:

```
my-MacBook:MuMoT user$ conda update conda
Solving environment: done

# All requested packages already installed.

my-MacBook:MuMoT user$ conda env create -n mumot-env -f environment_macos.yml 
Solving environment: failed

ResolvePackageNotFound: 
  - libiconv==1.15=0

```

I tried without success:

```
my-MacBook:MuMoT user$ conda install libiconv
Solving environment: done

# All requested packages already installed.

my-MacBook:MuMoT user$ conda env create -n mumot-env -f environment_macos.yml 
Solving environment: failed

ResolvePackageNotFound: 
  - libiconv==1.15=0
```

**Comment by willfurnass** [comment]
@joefresna What version of `libiconv` does `conda install libiconv` give you on macOS?

**Comment by joefresna** [comment]
`libiconv: 1.15-hdd342a3_7`

**Comment by willfurnass** [comment]
That's not available in the default repo:

```
$ conda search libiconv
Loading channels: done
# Name                  Version           Build  Channel             
libiconv                   1.14               0  pkgs/free           
libiconv                   1.14               0  conda-forge         
libiconv                   1.14               1  conda-forge         
libiconv                   1.14               2  conda-forge         
libiconv                   1.14               3  conda-forge         
libiconv                   1.14               4  conda-forge         
libiconv                   1.15      h470a237_1  conda-forge         
libiconv                   1.15      h470a237_2  conda-forge         
libiconv                   1.15      h470a237_3  conda-forge         
libiconv                   1.15      h63c8f33_5  pkgs/main     
```

Can you update the `environment.yml` so it requests `libiconv==1.15=4` and test again?

**Comment by willfurnass** [comment]
You could try one of:
 - removing all mention of `libiconv` from `environment_macos.yml`
 - Specifying `libiconv==1.14=0` in `environment_macos.yml`

**Comment by joefresna** [comment]
I specified `libiconv==1.14=0` and it worked.
I completed successfully all the installation steps, although Jupyter does not find the library:

```
ImportError                               Traceback (most recent call last)
<ipython-input-1-8e6381e0e006> in <module>()
----> 1 import mumot
      2 
      3 mumot.__version__

ImportError: No module named 'mumot'
```

**Comment by willfurnass** [comment]
Including step 7 on https://mumot.readthedocs.io/en/latest/install.html? i.e.

```sh
conda install graphviz
python -m pip install path/to/clone/of/MuMoT/repository
```

**Comment by joefresna** [comment]
sure.

```
(mumot-env) JF-MacBook:MuMoT joefresna$ conda install graphviz
Solving environment: done

# All requested packages already installed.

(mumot-env) JF-MacBook:MuMoT joefresna$ python -m pip install /Users/joefresna/MuMoT/
Processing /Users/joefresna/MuMoT
.....
<many lines removed>
.....
Successfully built mumot
Installing collected packages: mumot
  Found existing installation: mumot 0.1.0.dev0
    Uninstalling mumot-0.1.0.dev0:
      Successfully uninstalled mumot-0.1.0.dev0
Successfully installed mumot-0.1.0.dev0
```

**Comment by willfurnass** [comment]
Does the following work?
```sh
cd docs/
python -c 'import mumot; print(mumot.__file__)'
```

**Comment by joefresna** [comment]
is it a case-sensitive issue?

```
(mumot-env) JF-MacBook:docs joefresna$ python -c 'import mumot; print(mumot.__file__)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named 'mumot'


(mumot-env) JF-MacBook:docs joefresna$ python -c 'import MuMoT; print(MuMoT.__file__)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/joefresna/MuMoT/MuMoT/__init__.py", line 47, in <module>
    from mumot.process_latex.process_latex import process_sympy
ImportError: No module named 'mumot'
```

**Comment by willfurnass** [comment]
If `/Users/joefresna/MuMoT/` is your local repo then it should contain a `mumot` dir but no `MuMoT` dir if you've pulled down the latest changes to master.  What does `cd /Users/joefresna/MuMoT/ && git status` show?

**Comment by jarmarshall** [comment]
Hmmm... seems `MuMoT.py` isn't in the master branch - not sure why I can see a copy in my fork...

**Comment by jarmarshall** [comment]
Yes - just not sure how it got there as I though I was cloning the repository into an empty folder...

**Comment by jarmarshall** [comment]
@joefresna I have followed the installation instructions and, once the `libiconv` version number was resolved (please merge a pull request ASAP) I had no problems following the remaining installation instructions for conda and successfully running the notebook...

**Comment by jarmarshall** [comment]
PS we should get someone to check the Windows conda installation - maybe @renapagli or @ajc158

**Comment by joefresna** [comment]
@jarmarshall , it does still **not** work for me

@willfurnass , the command `git status` returns the following:

```
HEAD detached from 07d43bb
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   environment_macos.yml

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	.DS_Store
	.project
	.pydevproject
	COM3001_Demo.ipynb
	DemoForTalk.ipynb
	DemoForTalk_Brusselator.ipynb
	Demo_ConfCollBehaviour2018.ipynb
	Demo_ConfECMTB2018.ipynb
	MuMoT/MuMoT.py.orig
	SmallTest-Copy1.ipynb
	SmallTest-Copy1.ipynb.zip
	SmallTest-Copy2.ipynb
	SmallTest-Copy2.ipynb.zip
	SmallTest-Copy3.ipynb
	SmallTest.ipynb
	StrangeBugTest.ipynb
	model_bees.png
	thanks.png
	tmp.txt
```

I don't know where `MuMoT/MuMoT.py.orig` comes from... I never created it.

**Comment by willfurnass** [comment]
I think you're looking at an old commit: here's when commit `07d43bb` was created:
```
$ git --no-pager show -s  07d43bb
commit 07d43bb9cb4a1c6917b50bea2006594049ce6606
Author: James Marshall <james.marshall@sheffield.ac.uk>
Date:   Wed May 30 13:39:46 2018 +0100

    Fixes #16
```
I suggest you explicitly checkout the `master` branch then fetch and merge all newer commits from the master branch of this GitHub repo:
```
git checkout master
git fetch --prune --all --tags
git merge --ff-only
```

**Comment by joefresna** [comment]
oh... I am really sorry for this stupid mistake. :-( 
But I thought that `git pull origin master` would have been enough.

Anyhow, I tried with your commands to get the newer version and still I had the same identical error.

Then, by removing the directory and creating a new fresh clone, it finally worked!!
I guess the issue was that pulling it was not changing the subdir `MuMoT` into the current name `mumot`.

Thank you @willfurnass for the support.

I guess that we need to change `libiconv=1.14=0` in `environment_macos.yml` to close this issue.

**Comment by jarmarshall** [comment]
> I guess that we need to change libiconv=1.14=0 in environment_macos.yml to close this issue.

@joefresna I think that's you - if @renapagli is making us a new demo notebook based on his work however it would be good if he could test the Windows yml file...

**Comment by joefresna** [comment]
Trying with @renapagli on Windows,
`libiconv` needs to be 1.14
and 
`libxml2` needs to go to 2.9.4

to be updated in `environment_win.yml`

**Comment by jarmarshall** [comment]
I think when pull request #210 is merged this issue can be closed, as all the `.yml` files will have been removed.

**Comment by jarmarshall** [comment]
Closed by PR #210
